### DISCUSSION TITLE
Apache 2.0 license versus "All rights reserved"

### DISCUSSION THREAD

**Discussion by declann** [marimo-team/marimo#2676] [General]
Hi,

I tried marimo earlier this year and - working mostly from the reactive JS world - I was very happy to try reactive Python and impressed - congrats on what you've built!
Now I like the altair and charting improvements and I look forward to trying it out once more.

I noticed that this repo contains Apache 2.0 [LICENSE](https://github.com/marimo-team/marimo/blob/main/LICENSE) file, but some (key?) source code files are headed "Copyright 2024 Marimo. All rights reserved."

example: [main.py](https://github.com/marimo-team/marimo/blob/main/marimo/__main__.py)

Does this only feel incompatible to me?

Declan

**Comment by akshayka** [comment]
This was arrived off-thread, but the summary is that Apache 2.0 requires a copyright notice to be distributed with the project, which is what we do. Other projects do this too (for example, see TensorFlow).
